File: ICTCLAS50/ictclas.cpp
#include <boost/python.hpp>
#include <boost/python/list.hpp>
#include "ICTCLAS50.h"

using namespace boost::python;

bool ict_init(const char* pDirPath)
{
    return ICTCLAS_Init(pDirPath);
}

bool ict_exit()
{
    return ICTCLAS_Exit();
}

unsigned int import_dict(const char *sFilename,eCodeType codeType)
{
    return ICTCLAS_ImportUserDictFile(sFilename,codeType);
}

list process_str_ret_list(const char *sParag,int iLength,eCodeType codeType)
{
    int pResultCount=0;
    const tagICTCLAS_Result* re= ICTCLAS_ParagraphProcessA(sParag,iLength,pResultCount,codeType,false);
    list result;
    printf("the length is %d\n",pResultCount);
    for (int i=0;i<pResultCount;i++)
    {
        result.append(re[i]);
    }
    return result;
}

BOOST_PYTHON_MODULE(ictclas)
{
    class_<tagICTCLAS_Result>("tagICTCLAS_Result")
        .def_readonly("iStartPos",&tagICTCLAS_Result::iStartPos)
        .def_readonly("iLength",&tagICTCLAS_Result::iLength);
    enum_<eCodeType>("eCodeType")
        .value("UNKNOW",CODE_TYPE_UNKNOWN)
        .value("ASCII",CODE_TYPE_ASCII)
        .value("GB",CODE_TYPE_GB)
        .value("UTF8",CODE_TYPE_UTF8)
        .value("BIG5",CODE_TYPE_BIG5);

    def("ict_init",ict_init);
    def("ict_exit",ict_exit);
    def("import_dict",import_dict);
    def("process_str_ret_list",process_str_ret_list);
}

File: ICTCLAS50/Ictclas.py
# -*- coding: utf-8 -*-
import sys
import chardet
reload(sys)
sys.setdefaultencoding('utf-8')
import ictclas
#ictclas.import_dict('./userdict.txt',eCodeType.UTF8)
class Ictclas:
    def __init__(self,basepath='./'):
        ictclas.ict_init(basepath)

    def __del__ (self):
        ictclas.ict_exit()

    def split(self,s):
        #print '--------------split----------------'
        #print chardet.detect(s)
        length=len(s)
        #print s
        li=ictclas.process_str_ret_list(s,length,ictclas.eCodeType.UTF8)

        str=''
        for i in li:
            #print i.iStartPos
            str=str+s[i.iStartPos:(i.iStartPos+i.iLength)]+' '
        return str

if __name__=='__main__':
    c=Ictclas()
    words = c.split('中国你好')
    print words.split()

    for w in words.split():
        print w

